### DISCUSSION TITLE
Possible issue in RadialWaveFunction

### DISCUSSION THREAD

**Discussion by TanAndMichaelLover112** [nikolasibalic/ARC-Alkali-Rydberg-Calculator#171] [General]
Greetings,
I am unsure if this is my misuse or an error but if I try to calculate the transition rate from a high Rydberg state (70 p1/2) to a low one (ground, 5s1/2) for a Rubidium atom, I get a "TypeError: 'NoneType' object is not subscriptable". 
It appears that the span is too great for the code to handle as going from (n=) 70 to 60 works, however, I kind of need this jump
I am not too good on Python so you will have to forgive me if this is a simple fix.
Thank you,

Example Rubidium.getTransitionRate(70, 1, 1/2, 5, 0, 1/2)

If it's an issue due to calculation precision, is there a simple way to increase the precision?

**Comment by johnny-sa** [comment]
You need closed parenthesis when you use the class `Rubidium()`

i.e. Rubidium().getTransitionRate(70, 1, 0.5, 5, 0, 0.5)

which should return 

58.4454932620277

If 70 to 60 worked than it may be that you were not using the correct atom, as this would not be allowed in Caesium()

**Reply by TanAndMichaelLover112** [reply]
I copied your exact code and it gave me the same error I have been having. Is there something wrong with my version?

**Reply by nikolasibalic** [reply]
Do you have latest version of ARC 3.5.0 ?

If not, please update to latest version and re-try.

If yes, please share your Python version and your OS.

**Reply by TanAndMichaelLover112** [reply]
I tried it with 3.5.0 and master and still not working.
I'm using python 3.8.10 and Windows 11 version 10.0.22631 Build 22631
Not that I think it matters but I am also using Spyder

**Reply by dihm** [reply]
Also, could you post the full traceback for the error message? It should help narrow down where the error is coming from since we are having trouble reproducing the error you are seeing.

**Reply by TanAndMichaelLover112** [reply]
Here's my error message:

Rubidium().getTransitionRate(70, 1, 0.5, 5, 0, 0.5)
Traceback (most recent call last):

  Cell In[2], line 1
    Rubidium().getTransitionRate(70, 1, 0.5, 5, 0, 0.5)

  File C:\Users/name/OneDrive/Documents/Rydberg Calculations/Updated ARC/ARC-Alkali-Rydberg-Calculator-3.5.0\arc\alkali_atom_functions.py:1665 in getTransitionRate
    self.getReducedMatrixElementJ_asymmetric(

  File C:\Users/name/OneDrive/Documents/Rydberg Calculations/Updated ARC/ARC-Alkali-Rydberg-Calculator-3.5.0\arc\alkali_atom_functions.py:1138 in getReducedMatrixElementJ_asymmetric
    * self.getRadialMatrixElement(n1, l1, j1, n2, l2, j2, s=s)

  File C:\Users/name/OneDrive/Documents/Rydberg Calculations/Updated ARC/ARC-Alkali-Rydberg-Calculator-3.5.0\arc\alkali_atom_functions.py:939 in getRadialMatrixElement
    r1, psi1_r1 = self.radialWavefunction(

  File C:\Users/name/OneDrive/Documents/Rydberg Calculations/Updated ARC/ARC-Alkali-Rydberg-Calculator-3.5.0\arc\alkali_atom_functions.py:576 in radialWavefunction
    psi_r = d[0]

TypeError: 'NoneType' object is not subscriptable

**Reply by dihm** [reply]
It looks like you haven't installed ARC in the usual way. Is it possible you haven't compiled the numerov c extension? You can check that by changing your call to be `Rubidium(cpp_numerov=False).getTransitionRate(70, 1, 0.5, 5, 0, 0.5)`. Assuming that works, you'll need to properly install the arc package using pip instead of directly importing the source code. If you want to edit code, you can install in editable mode (see the pip documentation for details).

**Reply by TanAndMichaelLover112** [reply]
Ahh yes it works when I make that call. I went and reinstalled it using pip which seems to have fixed it.
Thank you so much for your patience!
